core/src/main/java/com/datastax/oss/driver/internal/core/context/DefaultDriverContext.java

```diff
@@ -361,20 +361,23 @@ protected EventBus buildEventBus() {
   @SuppressWarnings("unchecked")
   protected Compressor<ByteBuf> buildCompressor() {
     DriverExecutionProfile defaultProfile = getConfig().getDefaultProfile();
-    if (defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION)) {
-      String name = defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION);
-      if (name.equalsIgnoreCase("lz4")) {
+    if (!defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION)) {
+      return Compressor.none();
+    }
+
+    String name = defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION);
+    switch (name) {
+      case "lz4":
```

Comment:
The options were case-insensitive, let's preserve that behavior.
If you want to keep the switch statement, you can lower-case `name`.

Reply:
Good catch; that was a straight miss on my part. :(  Will fix, and I'll probably add some unit tests around that to validate going forward.

Reply:
One more thing I missed during my first review: config getters can take a default value, this simplifies this method a little bit. Addressing this directly in the merge.